src/highlighter.rs

```diff
@@ -40,16 +41,17 @@ impl Highlighter for DefaultHighlighter {
                     acc
                 }
             });
-            let buffer_split: Vec<&str> = line.splitn(2, &longest_match).collect();
+            let lowercase_buffer_split: Vec<&str> =
+                lowercase_line.splitn(2, &longest_match).collect();
+            let before_longest_match = line[..lowercase_buffer_split[0].len()].to_string();
+            let after_longest_match =
+                line[(lowercase_buffer_split[0].len() + longest_match.len())..].to_string();
 
-            styled_text.push((
-                Style::new().fg(self.neutral_color),
-                buffer_split[0].to_string(),
-            ));
+            styled_text.push((Style::new().fg(self.neutral_color), before_longest_match));
             styled_text.push((Style::new().fg(self.match_color), longest_match));
```

Comment:
Here I left longest_match but we can discuss if its better to put the real string with some uppercase character or not

Reply:
Do you have an example of what this will look like?

Reply:
```
From 1a882553028b51f11dca8676f0e0aabccd87724b Mon Sep 17 00:00:00 2001
From: Marco Zanrosso <mzanrosso@storvix.eu>
Date: Mon, 5 Jul 2021 23:09:12 +0200
Subject: [PATCH] Alternative output for the matched string

---
 src/highlighter.rs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/highlighter.rs b/src/highlighter.rs
index 9dd9ed7..96b1141 100644
--- a/src/highlighter.rs
+++ b/src/highlighter.rs
@@ -48,7 +48,12 @@ impl Highlighter for DefaultHighlighter {
                 line[(lowercase_buffer_split[0].len() + longest_match.len())..].to_string();

             styled_text.push((Style::new().fg(self.neutral_color), before_longest_match));
-            styled_text.push((Style::new().fg(self.match_color), longest_match));
+            styled_text.push((
+                Style::new().fg(self.match_color),
+                line[lowercase_buffer_split[0].len()
+                    ..(lowercase_buffer_split[0].len() + longest_match.len())]
+                    .to_string(),
+            ));
             styled_text.push((
                 Style::new().bold().fg(self.neutral_color),
                 after_longest_match,
--
2.28.0
```

Reply:
Seems like the PR should be a draft if you haven't tested it, yet. Please do definitely test before it lands.

Once you do, you can take a screenshot or copy/paste the output here. It'll help with reviewing.

Reply:
> Seems like the PR should be a draft if you haven't tested it, yet. Please do definitely test before it lands.
> 
> Once you do, you can take a screenshot or copy/paste the output here. It'll help with reviewing.

Only the example wasn't tested but I will drop here a before/after screenshot so you can easily understand this improvement.

Reply:
Before the PR changes, a command with some upper case characters, is not recognised:
<img width="75" alt="Schermata 2021-07-08 alle 09 43 07" src="https://user-images.githubusercontent.com/51322105/124884026-30d00f00-dfd2-11eb-8fcf-021d6037dd4d.png">

With this PR changes, the highlights is recognised even if you put some upper case characters:
![Registrazione schermo 2021-07-08 alle 09 45 49](https://user-images.githubusercontent.com/51322105/124883751-f2d2eb00-dfd1-11eb-8f31-e852731a582f.gif)

As you can see the command has been recognized and re-wrote all in a lower case, in one of my previous comment I have illustrated approximately how to rewrite the command without force it all in lower case.

Reply:
I wonder if this functionality should be part of the completer. If the user types `cLeA` and then another character, we probably want to just style that the best way that matches what the highlighter says.

Another way to say it is that we can separate the responsibility of the different parts of the system. The highlighter styles the code you type, the completer can update the code. We could have a second type of pass that does a kind of "autocorrect" pass also.

Reply:
Now that we have the completer yes I agree

Reply:
@jntrnr According to the latest changes with the Completer, we still want to continue for that road or do we keep this implementation?

Reply:
A related thing I think we could add is that when we complete, the completer could be fuzzy and match against the text even if the lowercase/uppercase didn't quite match.

So basically we wouldn't correct during the completion.

Curious to hear your thoughts

Reply:
@jntrnr Practically it is the same case that we are also facing here with the highlighter; surely this improvement will also be added to the completer.